toolkit/tools/pkg/imagecustomizerlib/customizeoverlay.go

```diff
@@ -68,40 +67,20 @@ func updateGrubConfigForOverlay(imageChroot *safechroot.Chroot, overlays []image
 	newArgs := fmt.Sprintf("rd.overlayfs=\"%s\"", concatenatedOverlays)
 
 	grubCfgPath := filepath.Join(imageChroot.RootDir(), "boot/grub2/grub.cfg")
-	lines, err := file.ReadLines(grubCfgPath)
+
+	grub2Config, err := file.Read(grubCfgPath)
 	if err != nil {
-		return fmt.Errorf("failed to read grub config: %w", err)
+		return fmt.Errorf("failed to read grub config:\n%w", err)
 	}
 
-	var updatedLines []string
-	linuxLineRegex, err := regexp.Compile(`^linux .*rd.overlayfs=.*`)
+	grub2Config, err = updateKernelCommandLineArguments(grub2Config, []string{"rd.overlayfs"}, newArgs)
```

Comment:
why both the array argument names and a formatted string of name/value pairs?
Would it make sense that this function takes an array where each element is a pair of strings?
`struct x { name, value}`

Reply:
To avoid parsing the input - since in most cases, I think, the caller can easily build the array of structures.

Reply:
Writing it this way is subtlety more powerful. It allows the args that are removed to be different from the args that are set. For example, when setting the SELinux mode, sometimes you want to add `enforcing=1` and sometimes you want to completely remove it.

Reply:
The args that are added should really be an array of strings, instead of a single combined string. I was just being lazy.